lib/core/src/lib/viewer/components/viewer.component.ts

```diff
@@ -300,7 +301,7 @@ export class ViewerComponent<T> implements OnDestroy, OnInit, OnChanges {
     ) {}
 
     ngOnChanges(changes: SimpleChanges) {
-        const { blobFile, urlFile } = changes;
+        const { blobFile, urlFile, mimeType } = changes;
 
         if (blobFile?.currentValue) {
             this.mimeType = blobFile.currentValue.type;
```

Comment:
Looking at this code, should we also handle getting mimeTypeIcon from the mimeType property as well? Is there a possibility of a case where the mimeTypeIcon shows a different icon compared to the value of the mimeType property?

Reply:
this component has no `mimeTypeIcon` property... we have a calculated `mimeTypeIconUrl` that gets the path to SVG file based on the mimeType

Reply:
I meant the mimeType input property mentioned at line num [209](https://github.com/Alfresco/alfresco-ng2-components/pull/10142/files/be65588bb874d7035a9592a59cc163debdf143c1#diff-8edc4da1fb1bec4de886be767b73f594e2c448e23cf2858260ba7f60acce88b3L209). Based on the code in line num 306 (where this comment thread has started), in addition to being changed via inputProperty from the parent, the mimeType variable will also change if the blobFile input property changes. However, the proposed code of updating the mimeTypeIcon only takes into account changes being made to the mimeType input property by the parent (via changes.mimeType). So I am wondering, if there can be a case, where only the blobFile property changes. The code here would also update the mimeType variable, but the mimeTypeIcon variable would not get updated (since the mimeType input property was not explicitly changed by the parent component)

Reply:
ok, got it, updating the url also on the blobFile change